tests/integration/test_delete.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from tests import RecordingSubscriber
+from tests.integration import BaseTransferManagerIntegTest
+from s3transfer.manager import TransferConfig
+
+
+class TestDeleteObject(BaseTransferManagerIntegTest):
+
+    def test_can_delete_object(self):
+        key_name = 'mykey'
+        self.client.put_object(Bucket=self.bucket_name,
+                               Key=key_name, Body=b'hello world')
+        self.assertTrue(self.object_exists(key_name))
+
+        transfer_manager = self.create_transfer_manager()
+        future = transfer_manager.delete(bucket=self.bucket_name,
+                                         key=key_name)
+        future.result()
+
+        self.assertFalse(self.object_exists(key_name))
+
+    def test_error_when_object_does_not_exist(self):
+        transfer_manager = self.create_transfer_manager()
+        future = transfer_manager.delete(bucket=self.bucket_name,
+                                         key='does_not_exist')
+        # The underlying DeleteObject API doesn't return any indication
+        # if you try to delete an object that does not exist.
+        result = future.result()
```

Comment:
You don't get an error from S3 is you delete an object that does not exist (that you otherwise have had access to).  I initially started writing the test intending to verify it propagated the exception, but realized that S3 doesn't return an error.

I've removed the test.